governance/default.go

```diff
@@ -1045,3 +1045,23 @@ func (gov *Governance) UseGiniCoeff() bool {
 func (gov *Governance) ChainId() uint64 {
 	return gov.ChainConfig.ChainID.Uint64()
 }
+
+func (gov *Governance) PendingChanges() map[string]interface{} {
+	return gov.changeSet.Items()
+}
+
+func (gov *Governance) Votes() []GovernanceVote {
+	return gov.GovernanceVotes.Copy()
+}
+
+func (gov *Governance) IdxCache() []uint64 {
+	gov.idxCacheLock.RLock()
+	defer gov.idxCacheLock.RUnlock()
+
+	return gov.idxCache
```

Comment:
`IdxCache()` should copy `idxCache`, like `PendingChanges()` and `Votes()`

Reply:
@ehnuje I agree that it is the way to go. But it requires the change on the type of idxCache and it can be pretty big. So I'm not sure it have to be done with this PR. I think the change of idxCache have to be in another PR to separate focus of each PR. May I have your opinion?

Reply:
@andybclee Wouldn't it be enough to copy `idxCache` while holding the lock like below? 
Please tell me if I missed something.

```
gov.idxCacheLock.RLock()
defer gov.idxCacheLock.RUnlock()

copiedCache := make([]uint64, 0, len(gov.idxCache))
copy(copiedCache, gov.idxCache)
return copiedCache

```

Reply:
@ehnuje Ah. I think I misunderstood your point. I think your mention is about to make idxCache to have methods such as Copy(), Size() and so on.

Reply:
@andybclee My bad =( I also agree with that my comment seems a bit confusing.